### PR TITLE
Convert GS Settings dialog to use wxwidgets

### PR DESCRIPTION
**Pull request by arcum42**
This scraps the old gtk GSdx dialog box, and replaces it with a wx version.

Current issues:
While I added in tooltips, they don't seem to actually be showing for some reason.
It still has the code for adding the pictures into a gtk resource and bundling them with the program, and the wx dialog does not use those pictures. Instead, it adds a logo if it finds one in the plugin folder. It's entirely possible we may want to remove the logo anyways, so not a major issue.

Otherwise, the dialog box should be hooked up and working, and I felt it was a good idea to go ahead and open the pr for it and get more eyes on it, though I would like to have the tooltip issue resolved before actually merging.

### CODE REVIEW

**Comment by arcum42** [comment]
Without logo:
gsdx-nologo

With logo:
gsdx-logo

**Comment by lightningterror** [comment]
With logo looks better, also for adapter maybe comment it out until it's used?

**Comment by tadanokojin** [comment]
I mentioned before in discord, but I prefer if we just got rid of the branding. It's kind of pointless for the direction we are going. Do you need help with the adapter stuff?

**Comment by arcum42** [comment]
With the logo, the current code shows the logo as in the second screenshot if it sees a copy of logo-ogl.bmp in the plugin folder, otherwise you get the first screenshot. 

If we don't want the logo, then leaving it as is would be fine. If we do, the choices would either be dropping a copy of that picture in that folder, or rewriting the resource code to pull it in to be wx, which will rather be a pain, though I'm sure the main program does that, so I could potentially figure out how to do it from that.

As far as the adapter goes, depends how far we're going to go with it. If we want to leave it as nonfunctional, I could disable it or hide it, I suppose. If we want to be able to pick a video card from it for OpenGL, that I could probably use help with. I mainly implemented it for later use if we replace the windows dialog box with it as well.

Still baffled by the fact the tooltips aren't working...

**Comment by tadanokojin** [comment]
Driver selects the adapter for GL so you might as well nop it out of Linux (VK can select an adapter later down the line). Really it's just DX. That's why I'm asking if you need my help on it since that's obviously going to be win only :P

**Comment by arcum42** [comment]
Well, making this dialog work in Windows is probably outside the scope of this pr. I was figuring I may go back later, and convert the various wx plugin dialog boxes to be used for windows as well in a separate pr or prs, and when we get to that point, I could probably use help on getting the adapters working.

**Comment by tadanokojin** [comment]
Fair enough, thought I'd ask since I saw the adapter code.

**Comment by arcum42** [comment]
Not saying I'd object if code to make the whole thing work in windows as well got dropped into the branch, mind you. :p 

Technically, I can work in windows, it just involves going to a different computer with a development environment I'm less comfortable with and only one monitor hooked up. I figure if I'm going to do that, might as well get it all out of the way at the same time...

**Comment by tellowkrinkle** [comment]
Working on macOS as well 
image

**Comment by arcum42** [comment]
Cool. TBH, I think it looks better in the Mac OS screenshot than in Linux...

**Comment by tadanokojin** [comment]
I took a quick look at windows and it looks like character encoding is going to make this a pain. I couldn't get wx to compile with multibyte (what gsdx currently uses) and it would require some work to get it to compile with unicode so maybe you are wise to hold off on it for a little bit.

File: plugins/GSdx/Window/GSwxDialog.cpp
Comment (tadanokojin) [inline plugins/GSdx/Window/GSwxDialog.cpp:23]: I think it would be good to make these helper functions static or use an anonymous namespace so it is unique to the translation unit

File: plugins/GSdx/Window/GSwxDialog.cpp
Comment (tadanokojin) [inline plugins/GSdx/Window/GSwxDialog.cpp:288]: You can do something like
```cpp
void HacksTab::CallUpdate(wxCommandEvent& /*event*/)
{
}
```

It'll save you a compiler warning. Same goes for anywhere else this occurs. It's a minor thing though so I'll leave it up to you.

File: plugins/GSdx/GS.cpp
Comment (tadanokojin) [inline plugins/GSdx/GS.cpp:48]: You don't really need the extra extern here. You already extern it in the header, so you can just include the header and you'll have the function. I prefer it that way anyway.

File: plugins/GSdx/Window/GSwxDialog.cpp
Comment (arcum42) [reply plugins/GSdx/Window/GSwxDialog.cpp]: I moved them to an anonymous namespace. Since my header has lots of "Type *name, *name2, *name3, *name4; ", trying to put the * on the Type starts looking fairly inconsistent. I just ran the files through clang-format, though, so they should adhere to any standards set in the format file now.

File: plugins/GSdx/Window/GSwxDialog.cpp
Comment (arcum42) [reply plugins/GSdx/Window/GSwxDialog.cpp]: Done.

File: plugins/GSdx/GS.cpp
Comment (arcum42) [reply plugins/GSdx/GS.cpp]: Taken care of, and I removed the commented old code while I was at it.

**Comment by arcum42** [comment]
The main blocker on this one is that the tooltips aren't working, btw. Not sure I actually have put tooltips in any wx dialog that's in a plugin before, though, so it could very well be that something with that's blocking it.

**Comment by tellowkrinkle** [comment]
I rebased and cleaned up the code a bit

Tooltips seem to work fine on macOS, have you checked to see if PCSX2's main UI's tooltips work for you?

**Comment by arcum42** [comment]
@tellowkrinkle 
Thanks!

Checked out the branch and did a clean build. Still seems like no tooltips. My main thought was that once GSdx is merged, we'll see how it does as far as tooltips with the merged version, because I've had this feeling it might have to do with being in a plugin and not part of the main program...

**Comment by arcum42** [comment]
Rebasing this branch seemed tricky enough that I just copied the changes to a wx-gs branch instead...

**Comment by arcum42** [comment]
Since the rebase has been taken care of, I'll delete the other branch I made.

Tooltips are sort of working half the time and not working the other half for me on Linux. At this point, I'd be fine with just moving forward with merging it if that's the only issue. I'm sure we'll rewrite without wx later anyways...

File: pcsx2/GS/Window/GSwxDialog.cpp
Comment (Florin9doi) [inline pcsx2/GS/Window/GSwxDialog.cpp:541]:
```suggestion
	: wxDialog(nullptr, wxID_ANY, "Graphics Settings", wxDefaultPosition, wxDefaultSize, wxCAPTION | wxCLOSE_BOX | wxRESIZE_BORDER)
```

**Comment by tadanokojin** [comment]
> At this point, I'd be fine with just moving forward with merging it if that's the only issue.

Gotta sneak a commit before that for making the adapter selection work on windows.

> Tooltips are sort of working half the time and not working the other half for me on Linux. 

I wanna pull them out of the resource header and inline them. Eventually I wanna get us setup for translations there but it can wait.

**Comment by arcum42** [comment]
Yeah, we'll want to take care of anything still needed for getting it to work properly on Windows first, and any other touchups that are needed. I'll be glad to see things out of that resource file, too...

File: pcsx2/GS/Renderers/DX11/D3D.cpp
Comment (tadanokojin) [inline pcsx2/GS/Renderers/DX11/D3D.cpp:24]: note to self

File: pcsx2/GS/Window/GSwxDialog.h
Comment (tadanokojin) [inline pcsx2/GS/Window/GSwxDialog.h:182]: note to self

**Comment by tadanokojin** [comment]
I added the adapter select but there's still win32 problems.

For one the tab text rendering is broken:
image

Additionally, this would remove a bunch of UX from the Windows config. Namely things are no longer contextually disabled anymore. For example it's possible now to enable hw hacks while the sw renderer is selected or to enable trilinear with d3d hw selected.

File: pcsx2/GS/Window/GSwxDialog.cpp
Comment (tellowkrinkle) [reply pcsx2/GS/Window/GSwxDialog.cpp:323]: Added

**Review by lightningterror — COMMENTED** [review]
Regarding the TC Offset hack, currently it is allowed to be changed on master and not greyed out on native res, rest are.

In Debug/OGL Geometry shader, Image load store, Sparse texture should be greyed out in Null and D3D.

Cosmetics:
Maybe the upper ones width can be increased to match the lower ones
image

I guess nothing can be done with the gap on d3d blending option?
image

For TC offsets, maybe a space between X: and the button, and Y: space between both buttons.

There are some other alignments or increasing with of spin buttons that can be done too but since QT is around the corner maybe they can be left out? For example in OSD increasing the width of timeout, size, max messages to match the sliders.

File: pcsx2/GS/Renderers/DX11/D3D.cpp
Comment (lightningterror) [inline pcsx2/GS/Renderers/DX11/D3D.cpp:1]: Missing copyright header.

File: pcsx2/GS/Renderers/DX11/D3D.h
Comment (lightningterror) [inline pcsx2/GS/Renderers/DX11/D3D.h:1]: Ditto.

File: pcsx2/GS/Window/GSwxDialog.cpp
Comment (lightningterror) [inline pcsx2/GS/Window/GSwxDialog.cpp:1]: Inconsistent copyright header, should be
https://github.com/PCSX2/pcsx2/blob/master/pcsx2/GS/GS.cpp

File: pcsx2/GS/Window/GSwxDialog.h
Comment (lightningterror) [inline pcsx2/GS/Window/GSwxDialog.h:1]: Ditto.

File: pcsx2/GS/Window/GSwxDialog.cpp
Comment (lightningterror) [inline pcsx2/GS/Window/GSwxDialog.cpp:242]: Would be nice to fix codacy warning too.

**Review by lightningterror — COMMENTED** [review]
Would be nice to fix the codacy warning, also the gl options sparse texture, geometry shader, image load store should be available on release builds, so it's better to hide the debug block portion instead of the whole tab.

**Comment by refractionpcsx2** [comment]
> Would be nice to fix the codacy warning, also the gl options sparse texture, geometry shader, image load store should be available on release builds, so it's better to hide the debug block portion instead of the whole tab.

I suggest rearranging the tab if it's gonna look weird if you cut it out. Debug stuff should probably be at the bottom anyway.

File: pcsx2/GS/Window/GSwxDialog.cpp
Comment (lightningterror) [inline pcsx2/GS/Window/GSwxDialog.cpp:343]: Can be marked as const, same for others applicable.

**Review by lightningterror — APPROVED** [review]
Aside from the minor tweaks that can be done, looks good.

**Review by JordanTheToaster — APPROVED** [review]

**Pull request merged**
